oauth2/runtime-flink-tests/build.gradle.kts

```
@@ -23,72 +23,188 @@ description = "Flink tests for Dremio AuthManager for Apache Iceberg"
+      runtimeConfig("org.apache.iceberg:iceberg-aws-bundle:$icebergVersion")
+    }
 
-  intTestImplementation(libs.flink.table.api.java)
-  intTestImplementation(libs.flink.table.runtime)
-  intTestImplementation(libs.flink.table.planner.loader)
-  intTestImplementation(libs.flink.clients)
-  intTestImplementation(libs.flink.connector.base)
-  intTestImplementation(libs.flink.connector.files)
+    // Create test task for this version combination
+    val testTask =
+      tasks.register<Test>(taskName) {
+        if (System.getenv("CI") == null) {
+          maxParallelForks = 2
+        }
 
-  intTestImplementation(libs.hadoop.common)
-  intTestImplementation(libs.hadoop.hdfs.client)
-  intTestImplementation(libs.hadoop.mapreduce.client.core)
+        group = "verification"
+        description =
+          "Runs Flink integration tests with Iceberg $icebergVersion and Flink $flinkVersion"
 
-  intTestRuntimeOnly("org.apache.iceberg:iceberg-aws")
-  intTestRuntimeOnly("org.apache.iceberg:iceberg-aws-bundle")
+        // Configure the test sources and classpath
+        testClassesDirs = sourceSets.intTest.get().output.classesDirs
+        classpath = runtimeConfig + sourceSets.intTest.get().output
 
-  intTestImplementation(platform(libs.testcontainers.bom))
-  intTestImplementation("org.testcontainers:testcontainers")
-  intTestImplementation(libs.s3mock.testcontainers)
+        // Enable JUnit Platform for test discovery
+        useJUnitPlatform()
 
-  intTestImplementation(platform(libs.junit.bom))
-  intTestImplementation("org.junit.jupiter:junit-jupiter")
-  intTestImplementation("org.junit.jupiter:junit-jupiter-api")
+        dependsOn(":authmgr-oauth2-runtime:shadowJar")
 
-  intTestImplementation(libs.assertj.core)
-  intTestImplementation(libs.mockito.core)
+        environment("AWS_REGION", "us-west-2")
+        environment("AWS_ACCESS_KEY_ID", "fake")
+        environment("AWS_SECRET_ACCESS_KEY", "fake")
+
+        jvmArgs(
+          "--add-exports",
+          "java.base/sun.nio.ch=ALL-UNNAMED",
+          "--add-opens",
+          "java.base/java.util=ALL-UNNAMED",
+          "--add-opens",
+          "java.base/java.lang=ALL-UNNAMED",
+          "--add-opens",
+          "java.base/java.lang.reflect=ALL-UNNAMED",
+          "--add-opens",
+          "java.base/java.io=ALL-UNNAMED",
+          "--add-opens",
+          "java.base/java.net=ALL-UNNAMED",
+          "--add-opens",
+          "java.base/java.nio=ALL-UNNAMED",
+          "--add-opens",
+          "java.base/java.util.concurrent=ALL-UNNAMED",
+          "--add-opens",
+          "java.base/java.security=ALL-UNNAMED",
+        )
+
+        // Set system properties to identify the versions being tested
+        systemProperty("authmgr.test.iceberg.version", icebergVersion)
+        systemProperty("authmgr.test.flink.version", flinkVersion)
+
+        // Use unique working directory and reports
+        workingDir = layout.buildDirectory.dir("tmp/$taskName").get().asFile
```

`.get().asFile` shouldn't be needed
```suggestion
        workingDir = layout.buildDirectory.dir("tmp/$taskName")
```